### PR TITLE
[Tag] Adds type=button attribute to Clickable Tag

### LINKED ISSUE #2894: [Tag component] Add the type=button attribute to avoid form submissions
## Feature request summary

Add the `type=button` attribute to Tag component to avoid submitting forms unexpectedly.

## Rationale

If the Tag component is used inside the Form component, when clicked it triggers a Form submit. Adding the `type="button"` will prevent that.
---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s [:+1: upvotes](https://help.github.com/articles/about-conversations-on-github/) to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md) for more information.

### PR DESCRIPTION
**Pull request by awebartisan**
Adds type=button attribute to Clickable Tag.

### WHY are these changes introduced?

Fixes #2894  

### WHAT is this pull request doing?
Adding `type="button"` attribute to clickable Tag.

### CODE REVIEW

**Review by chloerice — APPROVED** [review]
Thanks for catching this, @awebartisan!!

File: src/components/Tag/Tag.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ export function Tag({children, disabled = false, onClick, onRemove}: TagProps) {
   ) : null;
 
   const tagMarkup = onClick ? (
-    <button disabled={disabled} className={className} onClick={onClick}>
+    <button type="button" disabled={disabled} className={className} onClick={onClick}>
```
Comment (chloerice) [inline src/components/Tag/Tag.tsx:64]:
```suggestion
    <button 
      type="button"
      disabled={disabled}
      className={className}
      onClick={onClick}
    >
```

**Review by chloerice — DISMISSED** [review]
Just need you to sign the CLA and move the attributes on the `button` element to their own lines to fix the lint error causing the build to fail. If you happen to use VSCode as your text editor, add the Prettier extension and make any edit to the file then save and Prettier will autofix for you.

Screen Shot 2020-04-06 at 12 39 10 PM

**Comment by awebartisan** [comment]
Hi @chloerice, I made the commit before signing the CLA. Is there any way I can re-run the CLA check? 

Thanks.

**Comment by chloerice** [comment]
> Hi @chloerice, I made the commit before signing the CLA. Is there any way I can re-run the CLA check?
> 
> Thanks.

Hey @awebartisan! The linter is still complaining, looks like there's some extra whitespace after `<button `. If you pull down the commit made through the PR suggestion and fix that extra whitespace, all the checks will run again after you push up.

**Review by chloerice — APPROVED** [review]
Thank you so much for contributing @awebartisan 🙏!!

**Comment by chloerice** [comment]
Hey @awebartisan, everything is good to go but a few of the PR checks froze up. Would you mind pushing up from your branch again to re-trigger the checks?

**Comment by awebartisan** [comment]
Hey @chloerice , I would need to make some changes in order to be able to commit/push again. Can you suggest some?

Thanks 🙏🏼

**Comment by chloerice** [comment]
> Hey @chloerice , I would need to make some changes in order to be able to commit/push again. Can you suggest some?
> 
> Thanks 🙏🏼

Hey @awebartisan, you can resolve the conflicts in `UNRELEASED.md` 😊

**Comment by awebartisan** [comment]
Thank you 🙏🏼 I thought I messed up badly 😞Thank you for bearing with me. 

Looking forward to merge of my first pull request in Shopify ecosystem 😍

**Comment by juanpprieto** [comment]
Unfortunately this change makes the `type` prop to be hardcoded as "button".  Frameworks such as https://github.com/remix-run/remix use native form submit flows in creative ways to communicate between the client and the server in a full React context. Please allow the `type` prop to be overwritten when passed in.

Thanks!
JP

**Pull request merged**